internal/providers/terraform/aws/cloudfront_distribution.go

```diff
@@ -0,0 +1,67 @@
+package aws
+
+import (
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+)
+
+func GetCloudfrontDistributionRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "aws_cloudfront_distribution",
+		RFunc: NewCloudfrontDistribution,
+	}
+}
+
+func NewCloudfrontDistribution(d *schema.ResourceData, u *schema.ResourceData) *schema.Resource {
+	return &schema.Resource{
+		Name: d.Address,
+		SubResources: []*schema.Resource{
+			regionalDataOutToOrigin(u),
```

Comment:
Do we want another subresource for data transfer out to Internet and one for HTTP/HTTPS requests?

Reply:
Yes, my idea is to add each pricing factor as a sub resource because each of them has at least a few conditions.